elastic/kibana
# [ResponseOps] Move task retry determination to Task Manager

Resolves https://github.com/elastic/kibana/issues/151470

## Summary

Removes `getRetry` from the task type task type. I created a new error `throwRetryError` that can be thrown when we want to retry or retry at a specific date. I updated the actions framework to throw these new errors.

### Checklist



### To verify

- The retry logic should not be changed for a use. I would create a rule with an action and force the action to fail:
I usually update [Server Log](https://github.com/doakalexi/kibana/blob/main/x-pack/plugins/stack_connectors/server/connector_types/server_log/index.ts#L77) to throw an error here.

- Verify that the action does not retry when you just throw the error
- Update the return value [here](https://github.com/doakalexi/kibana/blob/main/x-pack/plugins/stack_connectors/server/connector_types/server_log/index.ts#L82) to have `retry: true` and verify that that the action will retry 3 times.
- Repeat the step above and set `retry: some date` and verify that the action will retry at that date. Some of the connectors already retry at a later date so you can also force an error in one of those to verify the retry logic.
